### DISCUSSION TITLE
How to access the data from all trials that meet a given criteria?

### DISCUSSION THREAD

**Discussion by viviers** [jspsych/jsPsych#717] [Q&A] [answered]
I need to access all previous trials where:

key_press = 'a' AND effort_level = 75

Then I need to create an array of their data for a variable called reward_offered (this variable takes integers)

So the array of all reward_offered might look like this:

array = [12, 14, 6, 9, 12];

**Comment by jodeleeuw** [comment, marked as answer]
Once you access the data with `jsPsych.data.get()` there are [a variety of operations](https://www.jspsych.org/overview/data/#aggregating-and-manipulating-jspsych-data) that can be used for filtering, extracting, summarizing, etc.

This is how you filter. Note that `key_press: 65` because the key code for `a` is `65`.
```js
var filtered_data = jsPsych.data.get().filter({key_press: 65, effort_level: 75})
```

Then you can extract `reward_offered` as an array (assuming that it is in the data for that trial):
```js
var array = filtered_data.select('reward_offered').values;
```

**Reply by viviers** [reply]
Thanks so much!
